libs/gpu-codec/src/main/java/org/elasticsearch/gpu/codec/ES92GpuHnswVectorsWriter.java

```diff
@@ -603,11 +603,15 @@ private void mergeByteVectorField(
                 // TODO: revert to CuVSMatrix.deviceBuilder when cuvs has fixed the multiple copies problem
                 var builder = CuVSMatrix.hostBuilder(numVectors, fieldInfo.getVectorDimension(), dataType);
 
+                IndexInput clonedSlice = slice.clone();
```

Comment:
Nit: maybe use try with resources, instead of explicitly closing

Reply:
Thanks, addressed in the last commit